src/core/cgroup.c

```diff
@@ -1808,6 +1835,30 @@ static bool unit_has_mask_realized(
                 u->cgroup_invalidated_mask == 0;
 }
 
+static bool unit_has_mask_disables_realized(
+                Unit *u,
+                CGroupMask target_mask,
+                CGroupMask enable_mask) {
+
+        assert(u);
+
+        return u->cgroup_realized &&
+                FLAGS_SET(u->cgroup_realized_mask, target_mask) &&
+                FLAGS_SET(u->cgroup_enabled_mask, enable_mask);
```

Comment:
if u->cgroup_realized is true, then the cgroup is not realized, hence we can say that all controllers are disabled, hence we can return true here, no?

Reply:
Hah, I never thought about it that way, but logically considering the way we use it, that makes sense.